### ISSUE TITLE
`createsuperuser` management command fails due to lack of username

### ISSUE THREAD

**Issue by Jongmassey** [opensafely-core/job-server#4076]
With a fresh install, [this line](https://github.com/opensafely-core/job-server/blob/main/DEVELOPERS.md?plain=1#L180) of teh developers guide fails due to the lack of prompting for the required non-null `username` field

**Comment by lucyb** [comment]
@ghickman to add details.

**Comment by ghickman** [comment]
When we added email-only users for OSIv2 we set `User.USERNAME_FIELD` to `"email"`, changing it from the default `"username"`.  Django uses this to construct the commands for the `createsuperuser` management command.  However, it doesn't then make use of this same configuration in `UserManager.create_superuser`, and advises that we should customise that to suit our own model.

We can fix this by writing our own `create_superuser` method on our `UserManager` that takes an email, [as per the Custom User docs](https://docs.djangoproject.com/en/5.0/topics/auth/customizing/#django.contrib.auth.models.CustomUserManager.create_superuser).  I suspect we can mirror [the built-in version](https://docs.djangoproject.com/en/5.0/topics/auth/customizing/#django.contrib.auth.models.CustomUserManager.create_superuser), dropping all the is_staff and is_superuser bits, and instead setting the CoreDeveloper role in `extra_fields` and leave the actual construction to `_create_user`.
